internal/counter.go

```diff
@@ -93,12 +82,60 @@ func (t *Counter) topAsSortedList() []*KeyCount {
 }
 
 func (t *Counter) GetTop() []*KeyCount {
-	t.lock.Lock()
-	defer t.lock.Unlock()
 	topList := t.topAsSortedList()
 	if len(topList) > t.size {
 		return topList[0:t.size]
 	} else {
 		return topList
 	}
 }
+
+// merge applies the counts from the SegmentCounter into the Counter.
+// Once merged, the SegmentCounter should be discarded.
+func (t *Counter) merge(r segmentCounter) {
```

Comment:
If I were writing this I'd use "segCounter" rather than just "r" for a variable name, but not a blocker. Also not sure about the "r_" prefix?

Reply:
I'll update these. also looks like r_foo isn't the regular go convention (downside of working in multiple languages)